### DISCUSSION TITLE
how to handle multi file progress

### DISCUSSION THREAD

**Discussion by mhamri** [Textualize/rich#3344] [Q&A]
I have two files each need to have its own progress. when I create a progress in each, I get error that one progress can be live at a time. what is the common pattern to handle these cases?

**Comment by hetari** [comment]
I am not sure if this will be useful to you or not, but I think the answer is in the [docs](https://rich.readthedocs.io/en/stable/progress.html#advanced-usage)

here instead of creating two `progress` at a time, you use one and add multiple tasks.

**Comment by mhamri** [comment]
but in that way I need to pass Progress instance down to the class. I wanted to be standalone somehow

**Reply by hetari** [reply]
I don't think `rich` can do that with the `Progress`, be sure to search the docs again (maybe you'll find something helpful), and search for other issues too.
